extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java

```diff
@@ -67,6 +64,19 @@ public class InfinispanDevServicesConfig {
     @ConfigItem
     public Optional<List<String>> artifacts;
 
+    /**
+     * List of the caches to be created and their associated cache templates.
+     * <p>
+     * To configure a cache, you would use one of the enum values from {@code org.infinispan.client.hotrod.DefaultTemplate}
+     * <p>
+     * quarkus.infinispan-client.devservices.cache-templates.cache1=DIST_SYNC
+     * quarkus.infinispan-client.devservices.cache-templates.cache2=REPL_SYNC
+     * <p>
+     * If an invalid cache template is passed, the Infinispan server will throw an error when trying to start.
+     */
+    @ConfigItem
+    public Map<String, String> cacheTemplates;
```

Comment:
Thanks for the contribution. I'm wondering if this is the right name, given it creates caches, not cache templates. Also wondering if it would be a good idea to be able to configure more things than just the template but I'm not familiar enough with Infinispan to be sure.

Let's ping @karesti to get her thoughts.

Reply:
Agreed on the name. New commit renamed to simply "caches".

Infinispan cache configurations can be extensive (https://infinispan.org/docs/stable/titles/configuring/configuring.html#cache-configuration_cache-configuration). However, for a large majority of devs, I think we just need a simple cache using one of the existing default templates.

Reply:
@stephennimmo those templates exist for a need of wildfly. They are indeed exposed, but it's a bad idea to use them. They don't configure any encoding, and the default encoding we want to be used in quarkus is protostream. I already saw your suggestion concerning the race condition, but tbh this is not the right solution from infinispan and quarkus perspective. we don't want to push the creation of caches through templates and we are working in a new API. 
I understand the race conditions concerns, but usually caches already exists in the server, or we can create them using the hotrod properties (application.properties has not yet been implemented to allow that). Having some caches created as you are implementing for the dev services is something we might want to consider, but not the use of templates since we are doing an effort to hide those from the user since it was a mistake to expose them in the first place. I don't know if I'm clear enough on this point.

Reply:
“They don't configure any encoding, and the default encoding we want to be used in quarkus is protostream”

I believe the DefaultTemplate enum is configured for protostream. 

“usually caches already exists in the server” 

That’s the point of the PR. When the devservices instance of infinispan is started, there are no caches created. There is a race condition. I can demo if needed.